File: Problems/Hirarchical Feature Flag System/Go/channel.go
package main

type Channel struct {
	id       string
	features map[string]bool
	parents  []string
}

func NewChannel(id string, features map[string]bool, parents []string) *Channel {
	return &Channel{
		id:       id,
		features: features,
		parents:  parents,
	}
}

func (c *Channel) SetFeature(featureFlag string, value bool) {
	c.features[featureFlag] = value
}

func (c *Channel) GetFeatures() map[string]bool {
	return c.features
}

func (c *Channel) GetParents() []string {
	return c.parents
}

File: Problems/Hirarchical Feature Flag System/Python/channel.py
class Channel:
    def __init__(self, id, features, parents):
        self.id = id
        self.features = features
        self.parents = parents

    def set_feature(self, feature_flag, value):
        self.features[feature_flag] = value

    def get_features(self):
        return self.features

    def get_parents(self):
        return self.parents